src/rez/package_maker__.py

```diff
@@ -128,6 +128,10 @@ def get_package(self):
     def _get_data(self):
         data = self._data.copy()
 
+        if "requires" in data:
```

Comment:
Sorry it's a little hard to track down the relevance of this change. What's going on here, and why would Nones be provided in the 'requires' key given to the package maker in the first place? Surely they should be stripped out beforehand?

Reply:
Hello Allan, this seems to be a really particular issue. I have only found one package that triggers it being importlib-metadata. It was a dependency of a package I was testing out (do not remember which one) but installing it directly still causes the issue.

```
Traceback (most recent call last):
  File "/home/iakritas/.rez/bin/rez/rez-pip", line 4, in <module>
    run('pip')
  File "/home/iakritas/.rez/lib/python2.7/site-packages/rez-2.35.0-py2.7.egg/rez/cli/_main.py", line 148, in run
    returncode = run_cmd()
  File "/home/iakritas/.rez/lib/python2.7/site-packages/rez-2.35.0-py2.7.egg/rez/cli/_main.py", line 140, in run_cmd
    return opts.func(opts, opts.parser, extra_arg_groups)
  File "/home/iakritas/.rez/lib/python2.7/site-packages/rez-2.35.0-py2.7.egg/rez/cli/pip.py", line 47, in command
    release=opts.release)
  File "/home/iakritas/.rez/lib/python2.7/site-packages/rez-2.35.0-py2.7.egg/rez/pip.py", line 482, in pip_install_package
    pkg.commands = '\n'.join(commands)
  File "/home/iakritas/.pyenv/versions/2.7.16/lib/python2.7/contextlib.py", line 24, in __exit__
    self.gen.next()
  File "/home/iakritas/.rez/lib/python2.7/site-packages/rez-2.35.0-py2.7.egg/rez/package_maker__.py", line 187, in make_package
    package = maker.get_package()
  File "/home/iakritas/.rez/lib/python2.7/site-packages/rez-2.35.0-py2.7.egg/rez/package_maker__.py", line 101, in get_package
    package_data = package_schema.validate(package_data)
  File "/home/iakritas/.rez/lib/python2.7/site-packages/rez-2.35.0-py2.7.egg/rez/vendor/schema/schema.py", line 195, in validate
    nvalue = Schema(svalue, error=e).validate(value)
  File "/home/iakritas/.rez/lib/python2.7/site-packages/rez-2.35.0-py2.7.egg/rez/vendor/schema/schema.py", line 235, in validate
    raise SchemaError([None] + x.autos, [e] + x.errors)
rez.vendor.schema.schema.SchemaError: Or(<class 'rez.utils.sourcecode.SourceCode'>, [Or(And(<type 'basestring'>, Use(<function expand_requirement at 0x7fb80733d320>)), And(<class 'rez.utils.formatting.PackageRequest'>, Use(<type 'str'>)))]) did not validate [u'zipp->=0.5', None, u'configparser->=3.5', None]
Or(Or(And(<type 'basestring'>, Use(<function expand_requirement at 0x7fb80733d320>)), And(<class 'rez.utils.formatting.PackageRequest'>, Use(<type 'str'>)))) did not validate None
Or(And(<type 'basestring'>, Use(<function expand_requirement at 0x7fb80733d320>)), And(<class 'rez.utils.formatting.PackageRequest'>, Use(<type 'str'>))) did not validate None
None should be instance of <class 'rez.utils.formatting.PackageRequest'>
```

Here are the contents of the data dictionary (notice the *requires* list):

```
{'commands': "env.PYTHONPATH.append('{root}/python')", 'description': u'Read metadata from Python packages', 'installed_variants': [], 'version': '0.18', 'skipped_variants': [], 'package_cls': <class 'rez.packages_.Package'>, 'requires': [u'zipp->=0.5', None, u'configparser->=3.5', None], 'variants': [['platform-linux', 'arch-x86_64', 'os-Arch-rolling', 'python-2.7']], 'name': u'importlib_metadata'}
```

That is why I have added the change, getting rid of None resolves the issue. I will dig a bit further and see if I can find anything else or any other packages that trigger the error.

Reply:
Looking at the importlib-metadata [setup.cfg](https://gitlab.com/python-devs/importlib_metadata/blob/master/setup.cfg) file I can see the install requires being as follows:

```
install_requires =
    zipp>=0.5
    pathlib2; python_version=='3.4.*' or python_version < '3'
    contextlib2; python_version < '3'
    configparser>=3.5; python_version < '3'

```
Now looking at the requires above:

```
'requires': [u'zipp->=0.5', None, u'configparser->=3.5', None]
```
It seems to be missing two of its dependencies? pathlib2 and contextlib2 should have been included.

Reply:
I think I've found the issue here. I'm going to address it in another PR, which aims to simplify code in pip.py by making further use of `packaging`, specifically for managing environment markers.

Reply:
Hello Allan, great news. If there is anything I can do to help please let me know. I will keep the PR open for now just in case.